app/src/main/java/tech/pauly/findapet/utils/BindingAdapters.java

```diff
@@ -36,32 +38,35 @@ public static void defaultSelectedItem(BottomNavigationView bottomNavigationView
 
     @BindingAdapter("imageUrl")
     public static void loadImageIntoView(ImageView view, String url) {
-        if (!TextUtils.isEmpty(url)) {
-            final Transformation transformation = new RoundedCornersTransformation(10, 0);
-            Picasso.with(view.getContext())
-                   .load(Uri.parse(url))
-                   .fit()
-                   .centerCrop()
-                   .transform(transformation)
-                   .into(view);
-        }
+        final Transformation transformation = new RoundedCornersTransformation(10, 0);
+        Picasso.with(view.getContext())
+               .load(Uri.parse(url))
+               .error(R.drawable.shape_animal_image)
+               .fit()
+               .centerCrop()
+               .transform(transformation)
+               .into(view);
     }
 
     @BindingAdapter("gridLayoutManager")
     public static void setGridLayoutManager(RecyclerView recyclerView, int span) {
-        GridLayoutManager layoutManager = new GridLayoutManager(recyclerView.getContext(), span);
-        GridItemDecoration dividerItemDecoration = new GridItemDecoration(recyclerView.getContext(), GridLayoutManager.HORIZONTAL);
-        recyclerView.setLayoutManager(layoutManager);
-        recyclerView.addItemDecoration(dividerItemDecoration);
+        if (span > 0) {
```

Comment:
Because sometimes we destroy the VM now when we teardown the view, so this value is defaulted to 0, thus, don't do any logic.